### PR TITLE
Fix npm version race condition

### PR DESCRIPTION
**Pull request by roryabraham**
This sets up a new Github Action that runs a node.js script. If we see the error encountered in the issue, then we'll:

1) Grab the npm patch version from `package.json`
2) Fetch all the tags from the RNC repo
3) Filter out tags from different patch versions
4) Grab the highest build version for which there's a tag in the repo
5) Update the npm version using that number + 1

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/147462

### Tests
**Note:** This is not tested yet, but my plan to test it it:

1) Create two dummy PR's (add some random markdown files)
2) Merge them both in quick succession. Hopefully this should reproduce the original issue
3) Merge this PR into master and wait for it to deploy.
4) Repeat steps 1 and 2, and hopefully the issue should be gone.

### CODE REVIEW

**Review by nickmurray47 — DISMISSED** [review]
LGTM pending Andrew's [change](https://github.com/Expensify/ReactNativeChat/pull/931/files#r543543498)

**Comment by AndrewGable** [comment]
Excited for this! Let me know when it's ready for a re-review.

**Comment by AndrewGable** [comment]
Bump

**Comment by roryabraham** [comment]
Okay, this has been updated.

Sort of tangentially related, but I spent a long time this weekend investigating automated testing of GH actions. Mostly I found that the tools aren't quite there yet and there are a lot of limitations, but I think it might be valuable enough to continue investigating. I was trying to get it set up something like this:

1) Use https://github.com/nektos/act and https://github.com/nektos/act-environments to run GH actions locally in a vagrant VM that mimics a real GH environment. Maybe pass information to that environment as necessary to emulate Expensify.cash in particular.
2) Use [nock](https://github.com/nock/nock) and/or [octokit-fixtures](https://github.com/octokit/fixtures) to mock the GH api for testing GH actions in the VM
3) Create wrapper scripts to run `act` for different event types and run assertions.

File: package.json [hunk 1]
```diff
@@ -25,6 +25,8 @@
     "detox-test": "detox test --configuration ios.sim.debug"
   },
   "dependencies": {
+    "@actions/core": "^1.2.6",
```
Comment (AndrewGable) [inline package.json:28]: Is it possible for these to be `devDependecies`?

Comment (roryabraham) [reply package.json:28]: I _think_ so. Do you know of a good way to know for sure?

Comment (AndrewGable) [reply package.json:28]: I thought it was if it's not used in the source, then it's not needed, but I am not sure off the top of my head which is installed when.

Comment (roryabraham) [reply package.json]: Yeah, I think these can be dev dependencies.

Comment (roryabraham) [reply package.json]: good call

**Review by AndrewGable — APPROVED** [review]

**Pull request merged**